### PR TITLE
Managed instance support TLS setting (Fixes #2261)

### LINKED ISSUE #2261: [Feature Request]: Add `minimalTlsVersion` to SQL Managed Instances
### Description

Similar to #1513 but for Managed Instances.

### PR DESCRIPTION
**Pull request by coolhome**
# Description

Implements SQL Managed Instances to have Minimum TLS version. This PR fixes #2261.

| Pipeline |
| - |
| |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings
- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (readme)
- [X] I did format my code

### CODE REVIEW

File: modules/Microsoft.Sql/managedInstances/deploy.bicep [hunk 1]
```diff
@@ -143,6 +143,15 @@ param encryptionProtectorObj object = {}
 @description('Optional. The administrator configuration.')
 param administratorsObj object = {}
 
+@allowed([
+  'None'
```
Comment (coolhome) [inline modules/Microsoft.Sql/managedInstances/deploy.bicep:147]: https://learn.microsoft.com/en-us/azure/templates/microsoft.sql/2022-05-01-preview/managedinstances?view=azuresql&pivots=deployment-language-bicep

**Comment by coolhome** [comment]
@microsoft-github-policy-service agree

**Review by AlexanderSehr — DISMISSED** [review]

File: modules/Microsoft.Sql/managedInstances/readme.md [hunk 2]
```diff
@@ -99,6 +99,7 @@ SQL MI allows for Azure AD Authentication via an [Azure AD Admin](https://docs.m
 | `vCores` | int | `4` |  | The number of vCores. Allowed values: 8, 16, 24, 32, 40, 64, 80. |
 | `vulnerabilityAssessmentsObj` | _[vulnerabilityAssessments](vulnerabilityAssessments/readme.md)_ object | `{object}` |  | The vulnerability assessment configuration. |
 | `zoneRedundant` | bool | `False` |  | Whether or not multi-az is enabled. |
+| `minimalTlsVersion` | string | `'1.2'` | `[None, 1.0, 1.1, 1.2]` | Minimal TLS version allowed. |
```
Comment (AlexanderSehr) [inline modules/Microsoft.Sql/managedInstances/readme.md:102]: Did you run the `Set-ModuleReadMe` script for this file? 

If not, please move the line a bit up to align with the alphabetical order. I think the script does that automatically and our pipelines will complain if it does not align

Comment (coolhome) [reply modules/Microsoft.Sql/managedInstances/readme.md]: @AlexanderSehr good catch, this is completed.

**Review by AlexanderSehr — APPROVED** [review]

**Comment by AlexanderSehr** [comment]
Approved and merged. Thank you very much for your contribution @coolhome 💪

**Pull request merged**